### PR TITLE
Publishing via Github actions

### PR DESCRIPTION
**Pull request by onebeastchris**
This ports the changes made in <https://github.com/GeyserMC/api/blob/feature/floodgate-merge/.github/workflows/build.yml> to the api/2.4.1 branch. 
Unlike https://github.com/GeyserMC/api/pull/7, this does not use all the new actions found in GeyserMC/actions - imo it makes little sense to use them (specifically, the release metadata/get version actions - given that the base api does not artifacts normally consumed by users, imo it makes little sense to use them just for the discord embed).

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - name: Get Version
+        if: ${{ github.repository == 'GeyserMC/api' }}
+        id: get-version
+        run: |
+          version=$(cat build.gradle | grep -o "version '[0-9\\.]*'" | cut -d" " -f2 | sed "s/'//g")
+          echo "VERSION=${version}" >> $GITHUB_OUTPUT
+
+      - name: Get Release Metadata
+        if: ${{ github.repository == 'GeyserMC/api' }}
+        uses: GeyserMC/actions/release@feature/globs
```
Comment (onebeastchris) [inline .github/workflows/build.yml]:
```suggestion
        uses: GeyserMC/actions/release@master
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        run: ./gradlew publish
+        env:
+          ORG_GRADLE_PROJECT_geysermcUsername: ${{ vars.DEPLOY_USER }}
+          ORG_GRADLE_PROJECT_geysermcPassword: ${{ secrets.DEPLOY_PASS }}
+
+      - name: Get Version
+        if: ${{ github.repository == 'GeyserMC/api' }}
+        id: get-version
+        run: |
+          version=$(cat build.gradle | grep -o "version '[0-9\\.]*'" | cut -d" " -f2 | sed "s/'//g")
```
Comment (Konicai) [inline .github/workflows/build.yml]: this must've been a copy/paste mistake, use what you had in #7 for this part

Comment (onebeastchris) [reply .github/workflows/build.yml]: oops, thanks - resolved it now

File: build-logic/src/main/kotlin/extensions.kt [hunk 3]
```diff
@@ -1,4 +1,12 @@
...
+import org.gradle.kotlin.dsl.the
 
-fun Project.isSnapshot(): Boolean =
-    version.toString().endsWith("-SNAPSHOT")
\ No newline at end of file
+fun Project.branchName(): String =
+    the<IndraGitExtension>().branchName() ?: System.getenv("BRANCH_NAME") ?: "local/dev"
+
+fun Project.shouldAddBranchName(): Boolean =
+    System.getenv("IGNORE_BRANCH")?.toBoolean() ?: (branchName() !in arrayOf("master", "local/dev"))
```
Comment (Konicai) [inline build-logic/src/main/kotlin/extensions.kt]: remove the IGNORE_BRANCH check

Comment (onebeastchris) [inline build-logic/src/main/kotlin/extensions.kt]:
```suggestion
fun Project.shouldAddBranchName(): Boolean =
    branchName() !in arrayOf("master", "main", "local/dev")
```

**Pull request merged**